File: hashing/first_letter_to_appear_twice.java
public char repeatedCharacter(String s)
{
    Set<Character> seen = new HashSet<>();
    for (int i = 0; i < s.length(); i++)
    {
        char c = s.charAt(i);
        if (seen.contains(c))
        {
            return c;
        }

        seen.add(c);
    }

    return ' ';
}

File: hashing/first_letter_to_appear_twice.js
/**
 * @param {string} s
 * @returns {character}
 */
var repeatedCharacter = function(s) {
    let seen = new Set();
    for (const c of s)
    {
        if (seen.has(c))
        {
            return c;
        }

        seen.add(c);
    }

    return " ";
}